added 1.49 release blog post (#2922)
FusionAuth/fusionauth-site: astro/src/content/blog/announcing-fusionauth-1-49.mdx

---
publish_date: 2024-03-04
title: Announcing FusionAuth 1.49
description: This update includes bug fixes, security improvements and more
authors: Dan Moore
image: /img/blogs/release-1-49/fusionauth-1-49.png
categories: Release Updates, Product
tags: release notes, linkedin, email, magic links, elasticsearch, opensearch
excerpt_separator: "{/* more */}"
---
import JSON from 'src/components/JSON.astro';
import Aside from '../../components/Aside.astro';

> FusionAuth version 1.49.0 shipped in early March, 2024. This version includes bug fixes, security improvements, and more.

Reviewer: ```suggestion
FusionAuth version 1.49.0 shipped in early March 2024. This version includes bug fixes, security improvements, and more.
```

{/* more */}

The focus of this update was bug fixes. We have some other bigger features planned soon, but wanted to get this release in the hands of the community.

Therefore, I hereby dub 1.49 the "Bug fix Beluga" release.

![Bug fixes are awesome.](/img/blogs/release-1-49/beluga-meme.png)

> All in all there are 27 issues, enhancements, and bug fixes included in the 1.49.0 release. As always, please see the [release notes](/docs/release-notes/#version-1-49-0) for a full breakdown of the changes between 1.48.3 and 1.49.0, including any schema changes.

Reviewer: ```suggestion
All in all, there are 27 issues, enhancements, and bug fixes included in the 1.49.0 release. As always, please see the [release notes](/docs/release-notes/#version-1-49-0) for a full breakdown of the changes between 1.48.3 and 1.49.0, including any schema changes.
```

Let's look at some of the more notable fixes.

## LinkedIn Provider Works Again

In 2023, LinkedIn changed their implementation of ["Sign in with LinkedIn"](https://learn.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin-v2). They made it more OIDC compliant (yay!). They also kept the old version around for older apps (yay!). But it broke FusionAuth's [LinkedIn Identity Provider](/docs/lifecycle/authenticate-users/identity-providers/social/linkedin) for new LinkedIn integrations (boo!).

The new version is what all new "Sign in with LinkedIn" integrations should use. The older version is [deprecated as of Aug 2023](https://learn.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin).

In this release, FusionAuth uses the new LinkedIn Authentication API which is OIDC compatible. This includes more standard OIDC scopes, such as `openid` and `profile`, as well as a more standardized Id token which can be validated using [LinkedIn's discovery document](https://www.linkedin.com/oauth/.well-known/openid-configuration).

> We're happy LinkedIn updated their authentication workflow to support OIDC and are excited to support that integration going forward.

Reviewer: ```suggestion
We're happy LinkedIn updated its authentication workflow to support OIDC and are excited to support that integration going forward.
```

## Elasticsearch Reindexing Progress

I don't know about you, but I like progress indicators. I remember, years ago, watching progress indicators all the time, from network downloads to disk defragmentation programs. (RIP `Defrag`.) We all have faster internet and disk now, so hopefully those indicators are a thing of the distant past for you.

> But if you use FusionAuth and [reindex your Elasticsearch (or Opensearch) index](/docs/lifecycle/manage-users/search/search#reindexing-elasticsearch), before this release FusionAuth didn't do a great job of reporting progress. It only reported back every 250k objects, which depending on the data in the index, could give the impression that progress has halted.

Reviewer: ```suggestion
But if you use FusionAuth and [reindex your Elasticsearch (or OpenSearch) index](/docs/lifecycle/manage-users/search/search#reindexing-elasticsearch), before this release FusionAuth didn't do a great job of reporting progress. It only reported back every 250k objects, which depending on the data in the index, could give the impression that progress has halted.
```

> In this release and going forward, FusionAuth will report reindexing progress more regularly in order to spare everyone some frustration.

Reviewer: ```suggestion
In this release and going forward, FusionAuth will report reindexing progress more regularly to spare everyone some frustration.
```

## Email Link Workflow Improvements

Speaking of frustration, FusionAuth has dealt with a lot of issues around [our magic links functionality](/docs/lifecycle/authenticate-users/passwordless/magic-links). This seems pretty straightforward:

* create a customizable, localizable email template: check!

> * create a one-time use, random, high entropy string tied to a user account: check!

Reviewer: Take or leave this one :D

```suggestion
* create a one-time use, random, high entropy string tied to a user account: ✅
```

Author: I'm going to pass on the emojis. I'm a curmudgeon, what can I say :).

> * create a one-time use, random, high entropy string tied to a user account: check!

Reviewer: ```suggestion
* create a one-time use, random, high entropy string tied to a user account: ✅
```

> * send the email to an account: check!

Reviewer: ```suggestion
* send the email to an account: ✅
```

If the user clicks on the link in the email, FusionAuth is assured they own the email address and should be logged in.  

But, it's not quite that simple.

> Link checkers, which are common in corporate email settings, defeat this in the name of preventing phishing. They retrieve all the links in the email to make sure they are safe, invalidating the one-time code.

Reviewer: I read this one like three times, I think this is what you are saying. Maybe this is more to the point? Or it is Tuesday and my 🧠 isn't working.

```suggestion
Corporate email link checkers prevent phishing attempts by automatically checking all links in an email. This means they access links before the user does, making one-time codes useless.
```

> FusionAuth tried to solve the email link checker issue by taking the GET request from the user clicking on a link in an email client, and returning an HTML form including the one-time code. After the form loads, FusionAuth submits it via JavaScript, which converts the `GET` into a `POST`. 

Reviewer: ```suggestion
FusionAuth tried to solve the email link checker issue by taking the GET request from the user clicking on a link in an email client and returning an HTML form including the one-time code. After the form loads, FusionAuth submits it via JavaScript, which converts the `GET` into a `POST`. 
```

In most cases, FusionAuth found web crawlers or link checkers will perform a `HEAD` request, perhaps followed by a `GET` request. If that is all the link checker does, the above strategy avoids one-time code invalidation. The user will get the desired result: they'll be logged in.

> However, FusionAuth clients have demonstrated that Outlook Safelinks both performs the `GET` and submits the form. This means that the `POST` request happens, which causes the one-time code to to be used before the user is presented with the page. The link is then invalid and when the user clicks it, they arrive at a screen indicating that.

Reviewer: ```suggestion
However, FusionAuth clients have demonstrated that Outlook Safelinks both performs the `GET` and submits the form. This means that the `POST` request happens, which causes the one-time code to be used before the user is presented with the page. The link is then invalid and when the user clicks it, they arrive at a screen indicating that.
```

> As far as the team could discover, Microsoft does not document any legitimate way to stop this or indicate that the link should not be followed and submitted. This makes sense because documented 'safe link' attributes or some other method of skipping the link check would allow a malicious actor to defeat them.

Reviewer: ```suggestion
As far as the team could discover, Microsoft does not document any legitimate way to stop this or indicate that the link should not be followed and submitted. This makes sense because documented `safe link` attributes or some other method of skipping the link check would allow a malicious actor to defeat them.
```

Reviewer: Did you want this highlighted?

Microsoft Outlook Safe Links is the most common offender, but not the only one. Others include:

> * ProofPoint URL Defense 

Reviewer: ```suggestion
* Proofpoint URL Defense 
```

> * mimecast

Reviewer: ```suggestion
* Mimecast
```
